AdminUi/test/AdminUi.Tests.Integration/API/BaseApi.cs

```diff
@@ -27,14 +31,14 @@ protected BaseApi(IOptions<HttpClientOptions> httpConfiguration, WebApplicationF
     private void LoadAndAddXSRFHeaders()
     {
         Task.Run(LoadXSRFTokensAsync).Wait();
-        _client.AddDefaultHeader(XSRF_TOKEN_HEADER_NAME, _xsrfToken);
-        _client.AddDefaultHeader("Cookie", $"{XSRF_TOKEN_COOKIE_NAME}={_xsrfCookie}");
+        _httpClient.DefaultRequestHeaders.Add(XSRF_TOKEN_HEADER_NAME, _xsrfToken);
+        _httpClient.DefaultRequestHeaders.Add("Cookie", $"{XSRF_TOKEN_COOKIE_NAME}={_xsrfCookie}");
     }
 
     private async Task LoadXSRFTokensAsync()
     {
-        var token = await Get<string>("/xsrf", new() { AcceptHeader = "text/plain" });
-        if (token.RawContent != null && token.Cookies != null && token.Cookies.Count > 0)
+        var token = await Get<string>("/xsrf", new RequestConfiguration { AcceptHeader = "text/plain" });
```

Comment:
Hmm, wouldn't it be easier if we manually executed this request instead of building so much cookie specific functionality into the `ExecuteRequest` method? We also don't need an `HttpResponse` object here. All we need is the value of the cookie.

Reply:
I agree, only reason I kept it this way was since it was already approved in a separate PR.
However, the usage of RestClient made things easier.